src/main/java/de/tum/in/www1/artemis/service/TextBlockService.java

```diff
@@ -49,23 +46,23 @@
      * @param submission TextSubmission to split
      * @return List of TextBlocks
      */
-    @Transactional(readOnly = true)
-    public List<TextBlock> splitSubmissionIntoBlocks(TextSubmission submission) {
+    public Set<TextBlock> splitSubmissionIntoBlocks(TextSubmission submission) {
+
+        // Return empty set for missing submission text.
         final String submissionText = submission.getText();
-        if (submissionText == null)
-            return new ArrayList<>();
-        // Return empty list for missing submission text.
+        if (submissionText == null) {
+            return Collections.emptySet();
```

Comment:
Does it make sense to use a immutable set here? The other returned set are mutable. Maybe refactor to only return immutable sets? `Set<String> immutable = ImmutableSet.copyOf(set);`